indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/PendingTaskBasedWorkerResourceManagementStrategy.java

```diff
@@ -0,0 +1,407 @@
+            workerConfig,
+            pendingTasks,
+            workers
+        );
+        while (want > 0) {
+          final AutoScalingData provisioned = workerConfig.getAutoScaler().provision();
+          final List<String> newNodes = provisioned == null ? ImmutableList.<String>of() : provisioned.getNodeIds();
+          if (newNodes.isEmpty()) {
+            log.debug("NewNodes is empty, returning from provision loop");
+            break;
+          } else {
+            currentlyProvisioning.addAll(newNodes);
+            lastProvisionTime = new DateTime();
+            scalingStats.addProvisionEvent(provisioned);
+            want -= provisioned.getNodeIds().size();
+            didProvision = true;
+          }
+        }
+      } else {
+        Duration durSinceLastProvision = new Duration(lastProvisionTime, new DateTime());
+        log.info("%s provisioning. Current wait time: %s", currentlyProvisioning, durSinceLastProvision);
+        if (durSinceLastProvision.isLongerThan(config.getMaxScalingDuration().toStandardDuration())) {
+          log.makeAlert("Worker node provisioning taking too long!")
+             .addData("millisSinceLastProvision", durSinceLastProvision.getMillis())
+             .addData("provisioningCount", currentlyProvisioning.size())
+             .emit();
+
+          workerConfig.getAutoScaler().terminateWithIds(Lists.newArrayList(currentlyProvisioning));
+          currentlyProvisioning.clear();
+        }
+      }
+
+      return didProvision;
+    }
+  }
+
+  private static Set<String> getWorkerNodeIDs(Collection<Worker> workers, WorkerBehaviorConfig workerConfig)
+  {
+    return Sets.<String>newHashSet(workerConfig.getAutoScaler().ipToIdLookup(
+        Lists.newArrayList(
+            Iterables.transform(
+                workers,
+                new Function<Worker, String>()
+                {
+                  @Override
+                  public String apply(Worker input)
+                  {
+                    return input.getIp();
+                  }
+                }
+            )
+        )
+    ));
+  }
+
+  int getScaleUpNodeCount(
+      final WorkerTaskRunnerConfig remoteTaskRunnerConfig,
+      final WorkerBehaviorConfig workerConfig,
+      final Collection<Task> pendingTasks,
+      final Collection<ImmutableWorkerInfo> workers
+  )
+  {
+    final int minWorkerCount = workerConfig.getAutoScaler().getMinNumWorkers();
+    final int maxWorkerCount = workerConfig.getAutoScaler().getMaxNumWorkers();
+    final Predicate<ImmutableWorkerInfo> isValidWorker = ResourceManagementUtil.createValidWorkerPredicate(config);
+    final int currValidWorkers = Collections2.filter(workers, isValidWorker).size();
+
+    // If there are no worker, spin up minWorkerCount, we cannot determine the exact capacity here to fulfill the need since
+    // we are not aware of the expectedWorkerCapacity.
+    int moreWorkersNeeded = currValidWorkers == 0 ? minWorkerCount : getWorkersNeededToAssignTasks(
```

Comment:
as you already pointed out minWorkers is adjusted below this. 
Yes, this check is still required since in case we dont have any workers, we cannot determine the worker capacity to determine how many more workers are needed.